src/lib/store/AuthModel.ts

```diff
@@ -177,31 +216,138 @@ export const getAuthModel = (): AuthModel => ({
 
     return false
   }),
-  signUp: thunk(async (actions, { email, password, name }) => {
-    const result = await actions.gravityUnauthenticatedRequest({
-      path: `/api/v1/user`,
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: {
+  signUp: thunk(async (actions, { email, password, name, accessToken, oauthProvider }) => {
+    let body
+    if (oauthProvider === "facebook") {
+      body = {
+        provider: "facebook",
+        oauth_token: accessToken,
+        email,
+        name,
+        agreed_to_receive_emails: true,
+        accepted_terms_of_service: true,
+      }
+    } else {
+      body = {
         email,
         password,
         name,
         agreed_to_receive_emails: true,
         accepted_terms_of_service: true,
+      }
+    }
+    const result = await actions.gravityUnauthenticatedRequest({
+      path: `/api/v1/user`,
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
       },
+      body,
     })
 
     // The user account has been successfully created
     if (result.status === 201) {
-      await actions.signIn({ email, password })
+      await actions.signIn({ email, password, accessToken, oauthProvider })
       actions.setState({
         onboardingState: "incomplete",
       })
       return true
     }
-    return false
+    return result
+  }),
+  authFacebook: thunk(async (actions) => {
+    const resultFacebook = await LoginManager.logInWithPermissions(["public_profile", "email"])
+    if (resultFacebook?.declinedPermissions?.includes("email")) {
+      Alert.alert("Error", "Allow the use of email to continue", [{ text: "Ok" }])
```

Comment:
```suggestion
      Alert.alert("Error", "Please allow the use of email to continue", [{ text: "Ok" }])
```

Do we have a guide for this kind of copy?

Reply:
no i guess